packages/bot-web-ui/src/components/journal.jsx

```diff
@@ -196,6 +196,27 @@ const JournalLoader = ({ is_mobile }) => (
     </ContentLoader>
 );
 
+const JournalItem = ({ row, is_new_row }) => {
```

Comment:
instead of is_new_row and the effect could isVisible from the react-virtualized rowRenderer be used? (if not it could be a hook as this logic is used for the other rows as well)

Reply:
Since it's a virtual list it removes and adds rows based on the scroll position. `is_new_row` is used here to track newly added row so that we can show transition only for the new row. Alternatively, we can skip transition during initial render and when it is scrolling.

Reply:
oh I see, could something like this be used and shared then?

```javascript
const useNewRowTransition = (is_new_row) => {
    const [in_prop, setInProp] = React.useState(!is_new_row);

    React.useEffect(() => {
        if (is_new_row) setInProp(true);
    }, [is_new_row]);

   return ({ in_prop })
}

const JournalItem = ({ row, is_new_row }) => {
  const { in_prop } = useNewRowTransition(is_new_row)
   // ...
}
```

Reply:
This is done now. Thanks for the suggestion 🙏